app/src/main/kotlin/com/simplemobiletools/notes/pro/fragments/ChecklistFragment.kt

```diff
@@ -24,93 +22,126 @@ import com.simplemobiletools.notes.pro.models.Note
-        view = inflater.inflate(R.layout.fragment_checklist, container, false) as ViewGroup
+        view   = inflater.inflate(R.layout.fragment_checklist, container, false) as ViewGroup
         noteId = arguments!!.getLong(NOTE_ID, 0L)
         return view
     }
 
     override fun onResume() {
         super.onResume()
 
-        NotesHelper(activity!!).getNoteWithId(noteId) {
-            if (it != null && activity?.isDestroyed == false) {
-                note = it
+        loadNoteById(noteId)
+    }
+
+    override fun setMenuVisibility(menuVisible: Boolean) {
+        super.setMenuVisibility(menuVisible)
+
+        if (menuVisible) activity?.hideKeyboard()
+    }
 
-                val checklistItemType = object : TypeToken<List<ChecklistItem>>() {}.type
-                items = Gson().fromJson<ArrayList<ChecklistItem>>(note!!.value, checklistItemType) ?: ArrayList(1)
-                if (config!!.moveUndoneChecklistItems) {
-                    items.sortBy { it.isDone }
+    private fun loadNoteById(noteId: Long) {
+        NotesHelper(requiredActivity).getNoteWithId(noteId) { storedNote ->
+            if (storedNote != null && activity?.isDestroyed == false) {
+                note = storedNote
+
+                try {
+                    val checklistItemType = object: TypeToken<List<ChecklistItem>>(){}.type
+                    items = Gson().fromJson<ArrayList<ChecklistItem>>(storedNote.value, checklistItemType) ?: ArrayList(1)
+                } catch (e: Exception) {
+                    migrateCheckListOnFailure(storedNote)
+                    if (isDebug) e.printStackTrace()
                 }
 
-                context!!.updateTextColors(view.checklist_holder)
+                if (config?.moveUndoneChecklistItems == true) items.sortBy { it.isDone }
+                requiredActivity.updateTextColors(view.checklist_holder)
                 setupFragment()
             }
         }
     }
 
-    override fun setMenuVisibility(menuVisible: Boolean) {
-        super.setMenuVisibility(menuVisible)
-        if (menuVisible) {
-            activity?.hideKeyboard()
+    private fun migrateCheckListOnFailure(note: Note) {
+        items.clear()
+
+        note.value.split("\n").map { it.trim() }.filter { it.isNotBlank() }.forEachIndexed { index, value ->
+            items.add(ChecklistItem(
+                id     = index,
+                title  = value,
+                isDone = false
+            ))
         }
+
+        saveChecklist()
     }
 
     private fun setupFragment() {
-        val plusIcon = resources.getColoredDrawableWithColor(R.drawable.ic_plus_vector, if (context!!.isBlackAndWhiteTheme()) Color.BLACK else Color.WHITE)
+        val plusIcon = resources.getColoredDrawableWithColor(R.drawable.ic_plus_vector, if (requiredActivity.isBlackAndWhiteTheme()) Color.BLACK else Color.WHITE)
+
         view.apply {
-            checklist_fab.apply {
+            with(checklist_fab) {
```

Comment:
avoid using "with" please, it is not used anywhere as it is quite confusing

Reply:
after consultation I use WITH because the value is not returned